test/DAppIntegration.t.sol

```diff
@@ -18,32 +19,42 @@ contract DAppIntegrationTest is Test {
     Atlas public atlas;
     MockDAppIntegration public dAppIntegration;
     DummyDAppControl public dAppControl;
+    AtlasVerification atlasVerification;
 
     address atlasDeployer = makeAddr("atlas deployer");
     address governance = makeAddr("governance");
     address signatory = makeAddr("signatory");
     address invalid = makeAddr("invalid");
 
     function setUp() public {
-    
         vm.startPrank(atlasDeployer);
-        // Computes the addresses at which AtlasVerification will be deployed
+
+        // Deploy ExecutionEnvironment with a salt
+        bytes32 salt = keccak256(abi.encodePacked(block.chainid, address(this), "AtlasFactory 1.0"));
+        ExecutionEnvironment execEnvTemplate = new ExecutionEnvironment{ salt: salt }(address(this));
```

Comment:
we don't need a salt when deploying the ExecutionEnvironment template/lib contract. The salt is only calculated in Factory, and is supposed to be a unique combo of user address x dapp control address x call config.

It should also take the Atlas contract address as an arg. address(this) refers to the address of the test contract (DAppIntegration.t.sol) which is fairly meaningless in this context. 

This test set up line can just be as follows:
```suggestion
        ExecutionEnvironment execEnvTemplate = new ExecutionEnvironment(expectedAtlasAddr);
```

And should ofc be moved to after `expectedAtlasAddr` is computed